bin/iron/src/app.rs

```diff
@@ -109,6 +109,72 @@ impl IronApp {
 
     #[cfg(not(target_os = "linux"))]
     fn build_menu() -> Menu {
+        let mut menu = Menu::new();
+
+        // app submenu
+        #[cfg(target_os = "macos")]
+        {
+            let app_name = "Iron";
+
+            menu = menu.add_submenu(Submenu::new(
+                app_name,
+                Menu::new()
+                    .add_native_item(MenuItem::About(
+                        app_name.to_string(),
+                        AboutMetadata::default(),
+                    ))
+                    .add_native_item(MenuItem::Separator)
+                    .add_item(
+                        CustomMenuItem::new("settings".to_string(), "Settings...")
+                            .accelerator("CmdOrCtrl+Comma"),
+                    )
+                    .add_native_item(MenuItem::Separator)
+                    .add_native_item(MenuItem::Services)
+                    .add_native_item(MenuItem::Separator)
+                    .add_native_item(MenuItem::Hide)
+                    .add_native_item(MenuItem::HideOthers)
+                    .add_native_item(MenuItem::ShowAll)
+                    .add_native_item(MenuItem::Separator)
+                    .add_native_item(MenuItem::Quit),
+            ));
+        }
+
+        // file submenu
+        let mut file_menu = Menu::new();
+        file_menu = file_menu.add_native_item(MenuItem::CloseWindow);
+        #[cfg(not(target_os = "macos"))]
+        {
+            file_menu = file_menu.add_native_item(MenuItem::Quit);
+        }
+        menu = menu.add_submenu(Submenu::new("File", file_menu));
+
+        // edit submenu
+        let mut edit_menu = Menu::new();
+        #[cfg(target_os = "macos")]
+        {
+            edit_menu = edit_menu.add_native_item(MenuItem::Undo);
+            edit_menu = edit_menu.add_native_item(MenuItem::Redo);
+            edit_menu = edit_menu.add_native_item(MenuItem::Separator);
+        }
+        edit_menu = edit_menu.add_native_item(MenuItem::Cut);
+        edit_menu = edit_menu.add_native_item(MenuItem::Copy);
+        edit_menu = edit_menu.add_native_item(MenuItem::Paste);
+        #[cfg(target_os = "macos")]
+        {
+            edit_menu = edit_menu.add_native_item(MenuItem::SelectAll);
+        }
+        menu = menu.add_submenu(Submenu::new("Edit", edit_menu));
+
+        // view submenu
+        #[cfg(target_os = "macos")]
+        {
+            menu = menu.add_submenu(Submenu::new(
+                "View",
+                Menu::new().add_native_item(MenuItem::EnterFullScreen),
+            ));
+        }
```

Comment:
can this be broken down into other functions? perhaps moving to a separate file to clean things up

only reason I'm asking is because all the `cfg` attributes can easily lead to us messing the build for the OS we're not using (like I did in the past by accident).
it's very unclear here what exists where

Reply:
It was copy pasted from [tauri's source code](https://docs.rs/tauri-runtime/0.14.0/src/tauri_runtime/menu.rs.html#245), so I was trying to keep it close to the original, but I'll break it down 👍

Reply:
I'll handle this separately